Compile Error of M2 - 1.20 on MacOS 10.15 due to givaro_bool_constant

Hello. I try to install Macaulay2 via Homebrew, but I have a compile error as below. My current MacAir has Catalina 10.15.7. I hope to resolve this issue. 

```
byeongsuyu@yubyeongsuui-MacBook-Air ~ % brew install Macaulay2/tap/M2
==> Downloading https://github.com/Macaulay2/M2/archive/refs/tags/release-1.20.t
Already downloaded: /Users/byeongsuyu/Library/Caches/Homebrew/downloads/b5f6b1a7be4a2589fca6b3cbf1ececa583f4bf433ffb236471bc18bab25aad25--M2-release-1.20.tar.gz
==> Installing macaulay2 from macaulay2/tap
==> Patching
==> git clone https://github.com/Macaulay2/M2-emacs.git M2/Macaulay2/editors/ema
==> git clone https://github.com/Macaulay2/memtailor.git M2/submodules/memtailor
==> git clone https://github.com/Macaulay2/mathic.git M2/submodules/mathic
==> git clone https://github.com/Macaulay2/mathicgb.git M2/submodules/mathicgb
==> cmake -GNinja -SM2 -BM2/BUILD/build-brew -DBUILD_NATIVE=OFF -DCMAKE_PREFIX_P
==> cmake --build M2/BUILD/build-brew --target M2-core M2-emacs M2-highlightjs
Last 15 lines from /Users/byeongsuyu/Library/Logs/Homebrew/macaulay2/06.cmake:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/mutex:190:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__mutex_base:14:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/chrono:829:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ratio:418:14: fatal error: no template named 'givaro_bool_constant'
    : public _LIBCPP_BOOL_CONSTANT((_R1::num == _R2::num && _R1::den == _R2::den)) {};
             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:450:36: note: expanded from macro '_LIBCPP_BOOL_CONSTANT'
#define _LIBCPP_BOOL_CONSTANT(__b) bool_constant<(__b)>
                                   ^
/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e/aring-zzp-ffpack.hpp:15:23: note: expanded from macro 'bool_constant'
#define bool_constant givaro_bool_constant
                      ^
1 error generated.
[264/406] /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DNDEBUG -DOM_NDEBUG -DSING_NDEBUG -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/d -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/c -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/BUILD/build-brew/Macaulay2/d -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e/mathic -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e/memtailor -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e/mathicgb -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/libxml2 -I/usr/local/opt/mpsolve/include -I/usr/local/opt/frobby/include -I/usr/local/opt/factory/include -I/usr/local/opt/flint/include -I/usr/local/opt/ntl/include -I/usr/local/opt/mpfi/include -I/usr/local/opt/mpfr/include -I/usr/local/opt/bdw-gc/include -I/usr/local/opt/gdbm/include -I/usr/local/opt/libatomic_ops/include -isystem /usr/local/opt/eigen/include/eigen3 -isystem /usr/local/opt/gmp/include -isystem /usr/local/Cellar/givaro/4.2.0/include -isystem /usr/local/Cellar/fflas-ffpack/2.5.0/include -isystem /usr/local/opt/libomp/include -isystem /usr/local/opt/tbb/include -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -Wuninitialized -g --sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/BUILD/build-brew/usr-host/include -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/include -I/tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/BUILD/build-brew/include -Wall -Wextra -Wfatal-errors -Wcast-qual -Wno-unused-parameter -Wno-attributes -Wno-unused-command-line-argument -Wno-cast-qual -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-register -Wno-mismatched-tags -Wno-unused-variable -Xpreprocessor -fopenmp /usr/local/opt/libomp/lib/libomp.dylib -Xclang -std=gnu++17 -MD -MT Macaulay2/e/CMakeFiles/M2-engine.dir/interface/aring.cpp.o -MF Macaulay2/e/CMakeFiles/M2-engine.dir/interface/aring.cpp.o.d -o Macaulay2/e/CMakeFiles/M2-engine.dir/interface/aring.cpp.o -c /tmp/macaulay2-20220615-26342-1d6ny7d/M2-release-1.20/M2/Macaulay2/e/interface/aring.cpp
ninja: build stopped: subcommand failed.

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/macaulay2/homebrew-tap/issues

These open issues may also help:
The Homebrew tap Macaulay2/tap/fflas-ffpack prevents Sage from building https://github.com/Macaulay2/homebrew-tap/issues/68
```

--- a/M2/Macaulay2/e/aring-zzp-ffpack.hpp
+++ b/M2/Macaulay2/e/aring-zzp-ffpack.hpp
@@ -9,6 +9,7 @@
 
 #include <type_traits> // define bool_constant to fix issue #2347
 #include <utility>
+#include <ratio> //fix compilation errors on some macs
 
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wconversion"

